cpp/ql/src/change-notes/2024-04-09-reduce-FP.md

```diff
@@ -0,0 +1,6 @@
+---
+category: majorAnalysis
+---
+* Reduce false positives of `GlobalUseBeforeInit.ql` and `InconsistentNullnessTesting.ql`.
+Many global variables that are initialized at declaration will be reported by `GlobalUseBeforeInit.ql`.
+When `checked` is in a macro expansion for handling exceptions, it is very likely for `InconsistentNullnessTesting.ql` to report false positives.
```

Comment:
I don't think `majorAnalysis` is quite right since this is modifying queries outside any of the security suites. So I suggest we change it to `minorAnalysis`.

I've also slightly reworded the description to align them with the wording we normally use. What do you think of these changes?

```suggestion
---
category: minorAnalysis
---
* The "Global variable may be used before initialization" query (`cpp/global-use-before-init`) no longer raises an alert on global variables that are initialized when they are declared.
* The "Inconsistent null check of pointer" query (`cpp/inconsistent-nullness-testing`) query no longer raises an alert when the guarded check is in a macro expansion.
```

Reply:
Fixed in https://github.com/github/codeql/pull/16149/commits/6cb5db2387a31447a7c9b98db9bee5acfc137aaf